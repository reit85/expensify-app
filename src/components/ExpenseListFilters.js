import 'react-dates/initialize'
//import DefaultTheme from 'react-dates/lib/theme/DefaultTheme'
import React from 'react';
import { connect } from 'react-redux';
import { DateRangePicker } from 'react-dates';
import { setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate } from '../actions/filters';

export class ExpenseListFilters extends React.Component {
  state = {
    calendarFocused: null
  }

  onDatesChange = ({ startDate, endDate }) => {
    this.props.setStartDate(startDate);
    this.props.setEndDate(endDate);
  }

  onFocusChange = (calendarFocused) => {
    this.setState(() => ({ calendarFocused }));
  }

  onTextChange = (e) => {
    this.props.setTextFilter(e.target.value);
  }

  onSortChange = (e) => {
    if (e.target.value === 'date') {
      this.props.sortByDate();
    } else if (e.target.value === 'amount') {
      this.props.sortByAmount();
    }
  }

  render() {
    return (
      <div className="page-filter">
        <div className="container">
          <div className="input-group">
            <div className="input-group__item">
              <input
                type="text"
                className="text-input"
                placeholder="search ..."
                value={this.props.filters.text}
                onChange={this.onTextChange}
              />
            </div>
            <div className="input-group__item">
              <select className="dropdown" value={this.props.filters.sortBy} onChange={this.onSortChange}>
                <option value="date">Date</option>
                <option value="amount">Amount</option>
              </select>
            </div>
            <div className="input-group__item">
              <DateRangePicker
                startDate={this.props.filters.startDate}
                endDate={this.props.filters.endDate}
                startDateId="startDateId"
                endDateId="endDateId"
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.calendarFocused}
                onFocusChange={this.onFocusChange}
                hideKeyboardShortcutsPanel={true}
                displayFormat="DD.MM.YYYY"
                showClearDates={true}
                numberOfMonths={1}
                isOutsideRange={() => false}
              />
            </div>
          </div>
        </div>
      </div>
    );
  }
};

const mapStateToProps = (state) => ({filters: state.filters})

const mapDispatchToProps = (dispatch) => ({
  setTextFilter: (text) => dispatch(setTextFilter(text)),
  sortByDate: () => dispatch(sortByDate()),
  sortByAmount: () => dispatch(sortByAmount()),
  setStartDate: (startDate) => dispatch(setStartDate(startDate)),
  setEndDate: (endDate) => dispatch(setEndDate(endDate))
})

export default connect(mapStateToProps, mapDispatchToProps)(ExpenseListFilters);
