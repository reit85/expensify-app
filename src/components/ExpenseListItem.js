import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment'
import numeral from 'numeral'

//moment.locale('de'); // Problem mit WEBPACK lÃ¤sst denn Test nicht zu ...

const ExpenseListItem = ({ id, description, amount, createdAt }) => (
  <div>
    <Link to={`/edit/${id}`}>
      <h3>{description}</h3>
    </Link>
    <p>
       {(amount/100).toLocaleString('de-DE', {style: 'currency', currency: 'EUR'})}
       - 
      {moment(createdAt).format('DD.MM.YYYY')}
     </p>
  </div>
);

export default ExpenseListItem;
