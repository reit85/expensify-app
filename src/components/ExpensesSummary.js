import React from 'react';
import { connect } from 'react-redux';
import { isNumber } from 'util';
import selectExpenses from '../selectors/expenses'
import selectExpensesTotal from '../selectors/expenses-total'

export const ExpensesSummary = ({expenseCount, expensesTotal}) => {
  const count = isNumber(expenseCount) ? expenseCount : 0
  const word = expenseCount == 1 ? 'expense' : 'expenses'
  const total = isNumber(expensesTotal) ? (expensesTotal/100) : 0
  const formatedExpensesTotal = total.toLocaleString('de-DE', {style: 'currency', currency: 'EUR'})
  return(
    <div>
      <h3>Viewing {count} {word} totaly {formatedExpensesTotal}.</h3>
    </div>
  )
}

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters)
  return {
    expenseCount: visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  }
}
 
export default connect(mapStateToProps)(ExpensesSummary)