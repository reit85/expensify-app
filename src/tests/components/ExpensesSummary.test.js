import React from 'react'
import {shallow} from 'enzyme'
import {ExpensesSummary} from '../../components/ExpensesSummary'

test('should render ExpensesSummary component without data', () => {
  const wrapper = shallow(<ExpensesSummary />)
  expect(wrapper).toMatchSnapshot()
})

test('should render summary with one expense', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={123}/>)
  expect(wrapper).toMatchSnapshot()
})

test('should render summary with many expenses', () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={123456789}/>)
  expect(wrapper).toMatchSnapshot()
})