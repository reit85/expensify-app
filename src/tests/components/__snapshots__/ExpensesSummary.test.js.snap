// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary component without data 1`] = `
<div
  className="page-header"
>
  <div
    className="container"
  >
    <h3
      className="page-header__title"
    >
      Viewing 
      <span>
        0
      </span>
       
      expenses
       totaly 
      <span>
        € 0.00
      </span>
      .
    </h3>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render summary with many expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="container"
  >
    <h3
      className="page-header__title"
    >
      Viewing 
      <span>
        23
      </span>
       
      expenses
       totaly 
      <span>
        € 1,234,567.89
      </span>
      .
    </h3>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render summary with one expense 1`] = `
<div
  className="page-header"
>
  <div
    className="container"
  >
    <h3
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totaly 
      <span>
        € 1.23
      </span>
      .
    </h3>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
